src/raw-transactions/sign-raw-transaction.js

```diff
@@ -1,34 +1,34 @@
 "use strict";
 
-var immutableDelete = require("immutable-delete");
 var signRawTransactionWithKey = require("./sign-raw-transaction-with-key");
-var isFunction = require("../utils/is-function");
+var RPCError = require("../errors/rpc-error");
+var errors = require("../errors/codes");
 var ACCOUNT_TYPES = require("../constants").ACCOUNT_TYPES;
 
 /**
  * Sign the transaction using either a private key or a signing function.
  * @param {Object} packaged Unsigned transaction.
  * @param {buffer|function} privateKeyOrSigner Sender's plaintext private key or signing function.
  * @param {string} accountType One of "privateKey", "uPort", or "ledger".
- * @param {function=} callback Callback function (optional).
+ * @param {function} callback Callback function.
  * @return {string} Signed and serialized raw transaction.
  */
 function signRawTransaction(packaged, privateKeyOrSigner, accountType, callback) {
-  try {
-    if (accountType === ACCOUNT_TYPES.PRIVATE_KEY) {
-      return signRawTransactionWithKey(packaged, privateKeyOrSigner, callback);
-    } else if (accountType === ACCOUNT_TYPES.LEDGER) {
-      privateKeyOrSigner(immutableDelete(packaged, "returns"), callback);
-    } else if (accountType === ACCOUNT_TYPES.U_PORT) {
-      privateKeyOrSigner(immutableDelete(packaged, "returns")).then(function (txHash) {
-        callback(null, txHash);
-      });
-    } else {
-      callback({ error: "unknown account type" });
-    }
-  } catch (error) {
-    if (!isFunction(callback)) throw error;
-    return callback(error);
+  switch (accountType) {
+    case ACCOUNT_TYPES.PRIVATE_KEY:
+      try {
+        return callback(null, signRawTransactionWithKey(packaged, privateKeyOrSigner));
+      } catch (err) {
+        return callback(err);
+      }
+    case ACCOUNT_TYPES.LEDGER:
+      return privateKeyOrSigner(packaged, callback);
+    case ACCOUNT_TYPES.U_PORT:
+      return privateKeyOrSigner(packaged).then(function (transactionHash) {
+        callback(null, transactionHash);
+      }).catch(callback);
+    default:
+      callback(new RPCError(errors.UNKNOWN_ACCOUNT_TYPE));
```

Comment:
It's not necessary in this case because it is the last statement in the function.  The other callbacks are "returned" simply to break out of the switch statement.  (`signRawTransaction` is async-only, so the callback will just have a void return.)

Reply:
Ok, makes sense.